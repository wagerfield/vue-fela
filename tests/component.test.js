import { createRenderer } from 'fela'
import {
  FelaRenderer,
  testSnapshot,
  installPlugin,
  wrapComponent
} from './test-helpers'
import component from './component.vue'

const localVue = installPlugin()
const fela = createRenderer()

describe('component.vue', () => {

  beforeEach(fela.clear)

  it('has $fela renderer property', () => {
    const wrapper = wrapComponent(component, localVue, fela)
    expect(wrapper.vm.$fela).toEqual(FelaRenderer)
    expect(wrapper.vm.$fela).toBe(fela)
  })

  it('renders expected snapshots', () => {
    const wrapper = wrapComponent(component, localVue, fela)

    testSnapshot(wrapper)

    wrapper.setProps({ active: true })
    testSnapshot(wrapper)

    wrapper.setProps({ margin: 2 })
    testSnapshot(wrapper)

    wrapper.setProps({ innerRadius: 2 })
    testSnapshot(wrapper)

    wrapper.setProps({ outerRadius: 4 })
    testSnapshot(wrapper)
  })
})
