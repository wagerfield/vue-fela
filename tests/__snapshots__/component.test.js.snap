// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.vue renders expected snapshots 1`] = `<div class="a b c d e f"></div>`;

exports[`component.vue renders expected snapshots 2`] = `<div class="a g c d e f"></div>`;

exports[`component.vue renders expected snapshots 3`] = `<div class="a g c d e f h"></div>`;

exports[`component.vue renders expected snapshots 4`] = `<div class="i g c d j k h"></div>`;

exports[`component.vue renders expected snapshots 5`] = `<div class="l g c m j k h"></div>`;
